### ISSUE TITLE
Revise Octavo `digest` structure

### ISSUE THREAD

**Issue by hauleth** [hauleth/octavo#45]
Currently I found that using `octavo::digest` is quite verbose, ex.:

```
use octavo::digest::Digest;
use octavo::digest::sha1::Sha1;
```

I am thinking about making all modules private and reexport hash functions as all of them currently has unique names (some ugly as `Sha3224`). But I am not so sure about it. What you guys think?

There are 3 options:
- left as is
- privatize and reexport
- left as is and simplify `Sha3XXX` names to `ShaXXX` (possible name collisions between `octavo::digest::sha2` and `octavo::digest::sha3`).

**Comment by Dr-Emann** [comment]
My personal preference would be as follows:

``` rust
use octavo::digest::Sha1;
use octavo::digest::sha2::Sha512;
use octavo::digest::sha3::Sha256;
```

This fits my expectations as a user: I've never heard of the the SHA3 versions referred to as - e.g. sha3512. SHA1 is its own algorithm, whereas SHA2/3 are families of algorithms. And it's still easy to use both in the same program:

``` rust
use octavo::digest::{sha2, sha3};
let digest1 = sha2::Sha512::default();
let digest2 = sha3::Sha512::default();
```

**Comment by hauleth** [comment]
Me too, but as SHA-3 is fairly new I do not hear much about it. SHA-2 functions are commonly refereed as `SHA-XXX` not `SHA2-XXX` and only tool that support SHA-3 under that name (that I know) is `rhash` which refers to it as `SHA3-XXX`.

To make it more weird and complicated SHA-1 and SHA-2 was described in the same document [FIPS 180-4](http://csrc.nist.gov/publications/fips/fips180-4/fips-180-4.pdf) and SHA-2 hash family was refereed as `SHA-XXX` there, this make impression that SHA-1 and SHA-2 (namely SHA-XXX) are related and SHA-3 is whole new hash function family.

**EDIT:**

Also [NIST on their page](http://csrc.nist.gov/groups/ST/toolkit/secure_hashing.html) refers to SHA-2 family as SHA-XXX and to SHA-3 as SHA3-XXX

**Comment by arthurprs** [comment]
@Dr-Emann suggestion looks good to me, when in doubt naming should be obvious even if we need one more level for the namespace.

**Comment by hauleth** [comment]
@arthurprs it also seems right to me, but I don't know if `Sha1` should be reexported. It could be both, but I am still not sure.

**Comment by hauleth** [comment]
This also leave us with whole MD family. It should be moved into one module named `octavo::digest::md` or should be left as is?

**Comment by hauleth** [comment]
FIxed by #53
